lib/azuremetadata/azuremetadatautils.py

```diff
@@ -0,0 +1,90 @@
+class QueryException(Exception):
+    pass
+
+
+class AzureMetadataUtils:
+    def __init__(self, data):
+        self._data = data
+        self._parents = {}
+        self._available_params = {}
+        self._parse_data(self._data)
+
+    @property
+    def available_params(self):
+        return self._available_params
+
+    def _parse_data(self, data, parent_key=''):
+        if isinstance(data, list):
+            for item in data:
+                assert isinstance(item, dict), "Only list of dicts is supported"
```

Comment:
This implies that if Msft changes the data format we blow up and we'd break registration potentially for weeks. This should be a soft failure.

Reply:
@rjschwei the concern is valid -- but we've agreed to set the API version to the oldest one by default (although we might need to override it in `/etc/regionsrvclient.conf` to get all the data we need for registrations); I think we can expect Msft not to break the format in existing API versions?

Reply:
OK, this is now a warning regardless.

Reply:
It is about future API versions, preferably we will not need to touch the code everytime a new revision of the api is released, The new revision just provides new data that shows up in the help and can be accessed.